middleware.ts

```diff
@@ -1,20 +1,33 @@
-import { NextRequest } from 'next/server'
-import { NextResponse } from 'next/server'
-
-export const middleware = (request: NextRequest) => {
-  // /articles will be shown in as URL but request will be against /articles/category/all
-  if(request.nextUrl.pathname === '/articles'){
-    return NextResponse.rewrite(new URL('/articles/category/all', request.url));
-  }
-
-  // Redirect to the /articles when manually type the /articles/category/all URL
-  if(request.nextUrl.pathname === '/articles/category/all'){
-    return NextResponse.redirect(new URL('/articles', request.url));
-  }
-  
-  return NextResponse.next();
-}
-
-export const config = {
-  matcher: ['/articles/category/all', '/articles'],
+import { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export const middleware = (request: NextRequest) => {
+  // /articles will be shown in as URL but request will be against /articles/category/all
+  if (request.nextUrl.pathname === '/articles') {
+    return NextResponse.rewrite(new URL('/articles/category/all/page/1', request.url));
+  }
+
+  // /articles will be shown in as URL but request will be against /articles/category/all
+  if (request.nextUrl.pathname === '/articles') {
+    return NextResponse.rewrite(new URL('/articles/category/all/page/1', request.url));
+  }
+
+  // Redirect to the /articles when manually type the /articles/category/all URL
+  if (request.nextUrl.pathname === '/articles/category/all') {
+    return NextResponse.redirect(new URL('/articles', request.url));
+  }
+
+  // /articles will be shown in as URL but request will be against /articles/category/all
+  if (/^\/articles\/category\/[^\/]+$/.test(request.nextUrl.pathname)) {

```

Comment:
We already have a similar behaviour for rewriting `/articles` to `/articles/category/all/page/1`. But in this case we also redirect from `/articles/category/all` to `/articles`. Do we also want to redirect from `/articles/category/all/page/1` to `/articles` and from `/articles/category/abc/page/1` to `/articles/category/abc`?